Maths/dotproduct.py

```diff
@@ -2,14 +2,15 @@
 The following is an implementation that takes to python lists and returns the
 dot product. If the lists are of different lengths or types, an error is printed.
 """
+from functools import reduce
 
 def dot_product(first_list, second_list):
-    dot_product = 0
     if(len(first_list) == len(second_list)):
         for i in range(0, len(first_list)):
```

Comment:
You are calculating via map and reduce in the 13th line then why are you using the for loop to iterate again and again and overwrite the value of dot_product?